OneCommunityGlobal/HighestGoodNetworkApp
# Nathan permissions merge

# Description
Fixes #6 (HIGH) Verify that permissions are implemented on the back-end and not just the front-end.
Permissions were split into front-end permissions and back-end permissions with some systems designed to accommodate this. This PR merges both permission sets into one so it's easier to make sure all permissions have relevant security checks implemented.

## Related PRS (if any):
This frontend PR is related to the #416 backend PR. 
https://github.com/OneCommunityGlobal/HGNRest/pull/416

## Main changes explained:
- I did a search and replace on all the front-end permissions to replace them with relevant back-end permissions.
- Deleted associatedPermissions.js and all references to it.
- Added Redux to hasPermission() calls to simplify arguments
- Memoized hasPermission() results

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Everything should work the same as before for every user.

## Screenshots or videos of changes:
Everything should look the same other than the code.

## Note:
If someone wants to help with unit testing, that would be appreciated. I'm considering it out of the scope of this PR though.
